packages/cli-hydrogen/src/cli/services/build.ts

```diff
@@ -3,50 +3,55 @@ import {ui, environment, error as kitError} from '@shopify/cli-kit'
 
 type Target = 'node' | 'client' | 'worker'
 
-interface BuildOptions {
+interface DevOptions {
   directory: string
   targets: {[key in Target]: boolean | string}
   base?: string
+  assetBaseURL?: string
   verbose?: boolean
 }
 
-async function build({directory, targets, base, verbose}: BuildOptions) {
+export function buildTaskList({directory, targets, base, assetBaseURL, verbose}: DevOptions): ui.ListrTask[] {
   const commonConfig = {base, root: directory}
 
-  const tasks: ui.ListrTask[] = Object.entries(targets)
+  return Object.entries(targets)
     .filter(([_, value]) => value)
-    .map(([key, value]) => {
-      return {
-        title: `Building ${key} code`,
-        task: async (_, task) => {
-          if (key === 'worker') {
-            process.env.WORKER = 'true'
-          }
-          try {
-            await viteBuild({
-              ...commonConfig,
-              build: {
-                outDir: `dist/${key}`,
-                ssr: typeof value === 'string' ? value : undefined,
-                manifest: key === 'client',
-              },
-              logLevel: verbose ? 'info' : 'silent',
-            })
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          } catch (error: any) {
-            const abortError = new kitError.Abort(error.message)
-            abortError.stack = error.stack
-            throw abortError
-          }
-
-          task.title = `Built ${key} code`
-        },
-      }
-    })
+    .map(([key, value]) => ({
+      title: `Building ${key} code`,
+      task: async (_, task) => {
+        if (key === 'worker') {
+          process.env.WORKER = 'true'
+        }
+        if (assetBaseURL) {
+          process.env.HYDROGEN_ASSET_BASE_URL = assetBaseURL
+        }
+
+        try {
+          await viteBuild({
```

Comment:
Mentioned in slack too, but thought I'd bring it up for discussion here. Thoughts on accepting a `buildCommand` similar to `oxygenctl` or maybe just running `yarn build` in general? Opens up a bit of flexibility for possibly projects that didn't use vite (which I think is a [possibility](https://github.com/Shopify/remix-oxygen/blob/main/package.json))

Reply:
I think it's a good idea to essentially let developers deploy "anything" with the command. This can be achieved in a couple different ways that will be documented in this [issue](https://github.com/Shopify/oxygen-platform/issues/571). For this particular PR we will move forward with the assumption that only Hydrogen is being used.